packages/react/src/primitives/Icon/icons/types.ts

```diff
@@ -0,0 +1,7 @@
+import { ViewProps } from '../../types';
+
+export type InternalIcon = (
+  props: ViewProps & {
+    size?: string | number;
```

Comment:
I think we have remove `size` prop on Icon long time ago. Why is this required?

Reply:
There is usage of the `size` prop with the internal `Icon` components throughout our codebase

Reply:
I think we should clean up the use because it takes no effect.

Reply:
Are we certain? I see it used to handle `height` and `width` [here](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/Icon/icons/IconClose.tsx#L16-L17). That being said, think that it would be fine to make changes around `size` since it's all internal, but should probably be handled in a separate PR

Reply:
Dang! I do not know why it was handled like this. That looks weird coz we should not lean on `size` any more and let the size scale based off parent font size